### DISCUSSION TITLE
atomWithObservable returns promise when there is no initial value

### DISCUSSION THREAD

**Discussion by edwardhpe** [pmndrs/jotai#1992] [Bug report]
I have two atoms created with `atomWithObservable`, where the second one depends on the first.
```
  const finalAtom = useMemo(
    () => {
      const firstAtom = atomWithObservable(
        () => getFirstObservable(id),
      );
      const secondAtom = atomWithObservable(
        (get) => {
          const firstValue = get(firstAtom);
          return getSecondObservable(firstValue.id);
        },
      );
      return atom((get) => ({
          first: get(firstAtom)?.data,
          second: get(secondAtom)?.data,
        }));
    },
    [id],
  );
```
The problem is this does not work because the `get(firstAtom)` in the getter for the second atom returns a `Promise` instead of the actual last value of the observable. I tried making the getter for the second observable `async` to await the promise but this is not supported for `atomWithObservable` apparently. This only works if I add `initialValue` options for the observable atoms. This is not desirable though because it removes my ability to leverage `Suspense`, and I have to add handling for the default values.

This is not a documented detail as far as I can tell. Is this defective, or intended behaviour? If it is intended behaviour, can it be documented?

**Comment by dai-shi** [comment]
Short answer is it's unsupported.

I think what would be ideal is if `getSecondObservable` accepts a promise as the argument and returns a new observable in sync. I'm not super familiar with observables or rxjs, but theoretically it sounds possible.

Otherwise, we have a similar issue in jotai-tanstack-query, and we worked around with atoms-in-atom pattern: https://github.com/jotaijs/jotai-tanstack-query/issues/21

**Comment by t6adev** [comment]
Closing this due to a lack of response from the owner for an extended period.
Please feel free to reopen if the issue persists.
